src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -3116,32 +3152,67 @@ private Message GetSortedSetCombineAndStoreCommandMessage(SetOperation operation
             {
                 SetOperation.Intersect => RedisCommand.ZINTERSTORE,
                 SetOperation.Union => RedisCommand.ZUNIONSTORE,
+                SetOperation.Difference => RedisCommand.ZDIFFSTORE,
                 _ => throw new ArgumentOutOfRangeException(nameof(operation)),
             };
             if (keys == null) throw new ArgumentNullException(nameof(keys));
+            // Difference operation can't have weights or aggregation
+            if (operation == SetOperation.Difference && (weights != null || aggregate != Aggregate.Sum))
+                throw new ArgumentException("ZDIFFSTORE cannot be used with weights or with aggregation.");
 
-            List<RedisValue>? values = null;
+            List<RedisValue>? values = new List<RedisValue>();
+            AddWeightsAndAggregation(ref values, weights, aggregate);
+            return new SortedSetCombineAndStoreCommandMessage(Database, flags, command, destination, keys, values?.ToArray() ?? RedisValue.EmptyArray);
+        }
+
+        private Message GetSortedSetCombineCommandMessage(SetOperation operation, RedisKey[] keys, double[]? weights, Aggregate aggregate, bool withScores, CommandFlags flags)
+        {
+            var command = operation switch
+            {
+                SetOperation.Intersect => RedisCommand.ZINTER,
+                SetOperation.Union => RedisCommand.ZUNION,
+                SetOperation.Difference => RedisCommand.ZDIFF,
+                _ => throw new ArgumentOutOfRangeException(nameof(operation)),
+            };
+            if (keys == null) throw new ArgumentNullException(nameof(keys));
+            // Difference operation can't have weights or aggregation
+            if (operation == SetOperation.Difference && (weights != null || aggregate != Aggregate.Sum))
+                throw new ArgumentException("ZDIFF cannot be used with weights or with aggregation.");
+
+            List<RedisValue>? values = new List<RedisValue>();
```

Comment:
See other comments - we should initialize this list with a size and it need not be nullable :) van simplify to `var values` here. In fact, instead of a list let's just create the array directly here and copy in `for` loop style. Again, happy to push changes intended but don't want to step on toes and goal is to get on the same page!

Reply:
Yea maybe I need your help with this part :sweat_smile: 
I am confused because I want to use list to reuse the method that added the weights and the aggregation.

Reply:
Took another approach here - no longer relevant! Avoids the duplicate allocations to only the array (if any) we're going to ref anyhow,